fw/hpack.c

```diff
@@ -1191,13 +1191,17 @@ tfw_hpack_hdr_set(TfwHPack *__restrict hp, TfwHttpReq *__restrict req,
 done:
 	switch (entry->tag) {
 	case TFW_TAG_HDR_H2_METHOD:
+		/* If @hp->index corresponds to existing entry in HPACK
+		 * static table, set http method according to that index.
+		 * Otherwise the entry is in dynamic table and we need
+		 * to parse the method here.
+		 */
 		if (hp->index == 2) {
 			req->method = TFW_HTTP_METH_GET;
 		} else if (hp->index == 3) {
 			req->method = TFW_HTTP_METH_POST;
 		} else {
-			WARN_ON_ONCE(1);
-			return T_DROP;
+			tfw_idx_hdr_parse_method(req, entry->hdr);
```

Comment:
I think the change is incorrect and dangerous, see the comment about the test case from #1617 and `h2spec`. `h2spec`, `curl` and browsers like Firefox or Chrome are the good references to check which cases we should support.

By the way, it makes sense to run `h2spec` before the PR and with the PR and compare whether some issues are gone or some new issues were introduced. Let's run the tool for all h2 specific changes to make sure that we're on the right track. We can't switch the test on for CI because at the moment (with this PR) we have `146 tests, 126 passed, 1 skipped, 19 failed`, so several tests are failing. At some point we need to fix all the issues besides the issues about the features, which are in the TODO/doubtful list, e.g. server PUSH. @const-t FYI also.